### PR TITLE
Upstream sync Kong v2.38.0

### PR DESCRIPTION
**Pull request by ced0ps**
This PR aligns with upstream charts release 2.38.0 and includes a few additional commit up to [07ddc3d](https://github.com/Kong/charts/compare/kong-2.38.0...07ddc3d) 

## Checklist

- [x] Automated test are working (for chart changes)
- [x] Changelog entry has been added

### Manual tests on workload clusters

For plain installations (default values) and database mode (deploy `tests/manual/postgres.yaml`, then install with `tests/manual/values-database.yaml`), execute these tests. If you have additional configuration, make sure your existing deployments with custom values still work.

- [x] Upgrade from previous version works
- [x] Existing Ingress resources are reconciled correctly (change domain, see if its available)
- [x] Fresh install works
- [x] Fresh Ingress resources are reconciled correctly

### CODE REVIEW

File: helm/kong-app/values.yaml [hunk 1]
```diff
@@ -718,7 +725,7 @@ postgresql:
     repository: giantswarm/postgresql-bitnami
     # use postgres < 14 until is https://github.com/Kong/kong/issues/8533 resolved and released
     # enterprise (kong-gateway) supports postgres 14
-    tag: 13.6.0-debian-10-r52
+    tag: 13.11.0-debian-11-r20
```
Comment (ubergesundheit) [inline helm/kong-app/values.yaml:728]:
```suggestion
    tag: 13.6.0-debian-10-r52
```

Comment (ubergesundheit) [reply helm/kong-app/values.yaml:728]: This should also fix the errors in CI (Image not available)

File: helm/kong-app/CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,44 @@
 
 ## Unreleased
 
-Nothing yet.
+### Changes
+
+* Updated handling of `session_conf_secret` to accommodate Kong 3.6.
```
Comment (ubergesundheit) [inline helm/kong-app/CHANGELOG.md:7]: Now that I see it, technically we're syncing ahead of 2.38.0

I suggest changing the entries in CHANGELOG.md and README.md to point to the commit you've synced to

Comment (ced0ps) [reply helm/kong-app/CHANGELOG.md:7]: Yea, I think I realized that when checking for the CRD changelogs. I'll make the changes.

**Review by ubergesundheit — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 ### Changes
 
-- Align with upstream chart version [2.38.0](https://github.com/Kong/charts/releases/tag/kong-2.38.0) ([Changes in upstream repository](https://github.com/Kong/charts/compare/kong-2.35.1...kong-2.38.0))
+- Align with upstream chart version [07ddc3d](https://github.com/Kong/charts/releases/tag/kong-2.38.0) ([Changes in upstream repository](https://github.com/Kong/charts/compare/kong-2.35.1...07ddc3d))
```
[the remark this answers is not in the document]
Comment (ced0ps) [reply CHANGELOG.md:12]: Should it point to the specific commit instead? I feel the release ref has more value. Or is it confusing?

[the remark this answers is not in the document]
Comment (ubergesundheit) [reply CHANGELOG.md:12]: I would expect to be taken to a commit and it takes me to a release. That would confuse me. If you think the link needs more explanation, maybe use `[sha](link to sha) (ahead of [release x](link to release))` or something similar

**Pull request merged**